### PR TITLE
Update various sending commands to return a value

### PR DESCRIPTION
**Pull request by Wolfsblvt**
### Changes
- Most send commands return the message being send now
  - `/sendas`
  - `/send`
  - `/sys`
  - `/comment`
  - `/ask` did already work this way
  - Can't be changed: `/trigger`, because by default it does not await the message generation
- Refactored checks on `/sendas` at the beginning, as they were very redundant
- Updated `/ask` toasts to warnings, being more consistent with the other commands.
  Errors should be reserved for something unexpectedly being wrong or actually failing. Missing or falsely formatted args can just be a warning.
- Function `sendMessageAsUser` now has a return value of the message object itself. Does not break existing code.

#### Possible "Breaking" Changes
`/sendas name={{char}} Some message beginning | /continue`
will not work anymore. It does not error out either, it'll silently pass the message as a custom prompt into the continue.
Solution is easy though, people should get used to using pipe breakers.
This works: `/sendas name={{char}} Some message beginning || /continue`

## Checklist:

- [x] I have read the [Contributing guidelines](https://www.youtube.com/watch?v=dQw4w9WgXcQ).

### CODE REVIEW

**Comment by Succubyss** [comment]
I'm personally in favor. However, if this is refused, let the ability to have it return a value at least become a parameter like `emit=` or `pass=`

**Review by Cohee1207 — CHANGES_REQUESTED** [review]
This is a breaking change and I'm not a fan of it. This has limited usability (only ever usable if you have specific message-altering regex), but high muscle memory breakage potential. I'm ready to take a Luccy pill and suggest to enable this behavior with a named argument.

**Comment by Wolfsblvt** [comment]
What would that one be named?
And should be update existing commands that are already returning to so this as well?

I think `return=` is the most reasonable actually, that we can consistently use across all commands in the future, without conflicts or being weirdly named for the command.
`output=`, like the current `listinjects` might be an option too, but I have a feeling that might go against some commands where output does not make much sense.

Values could be:
- none
- pipe
- toast
- console

Dunno if the options "chat" and "popup" would globally make sense, that one applies mostly to injects.

**Comment by Cohee1207** [comment]
> What would that one be named?

"return" is fine.

> And should be update existing commands that are already returning to so this as well?

Yes, but it's only /ask so far? + potential aliases and refactoring for /listinjects and /listvar

> output=, like the current listinjects might be an option too, but I have a feeling that might go against some commands where output does not make much sense.

Actually it's "format", not "output".  If you think this name sucks, let's add a couple aliases. And extract common function for doing the "return" thing.

**Comment by Wolfsblvt** [comment]
> Actually it's "format", not "output". If you think this name sucks, let's add a couple aliases. And extract common function for doing the "return" thing.

Working on my fix/update. Gimme a bit. Though just wanted to reply, aliases for named aren't really well supported in the current implementation really. You'd have to build them yourself in the callback handler, which can be quite confusing. But it certainly is an option. Len said he does not want to have those though.

**Comment by Wolfsblvt** [comment]
I am too tired to update the PR description or summarize the stuff in those commits. Read the commit messages and details you can expand :/
It done though.

File: public/scripts/slash-commands/SlashCommandReturnHelper.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        const stringValue = typeof value !== 'string' ? actualConverterFunc(value) : value;
+
+        switch (type) {
+            case 'popup-html':
+            case 'popup-text':
+            case 'chat-text':
+            case 'chat-html':
+            case 'toast-text':
+            case 'toast-html': {
+                const htmlOrNotHtml = shouldHtml ? (new showdown.Converter()).makeHtml(stringValue) : escapeHtml(stringValue);
```
Comment (Cohee1207) [inline public/scripts/slash-commands/SlashCommandReturnHelper.js]: `/sendas return=toast-html name=Stelle <script>alert('pwned')</script>`

**Comment by Cohee1207** [comment]
`/listinjects` doesn't work when there are no injects in the chat. In staging it makes a toast that the chat has no injects.
image

File: public/scripts/slash-commands/SlashCommandReturnHelper.js (hunk 1 above)
Comment (Wolfsblvt) [reply public/scripts/slash-commands/SlashCommandReturnHelper.js]: Done. I had it in there before. Dunno where along the way it went poof.

**Comment by Wolfsblvt** [comment]
> `/listinjects` doesn't work when there are no injects in the chat. In staging it makes a toast that the chat has no injects.

You think a toast is the best way if you want pipe or object as return? But it should not crash for sure.

Edit: Fixed.

**Comment by Cohee1207** [comment]
~~`format` in `classify-expressions` is not deprecated. It defines the actual piped value. I'm fixing that.~~

UPD: !jar

**Review by Cohee1207 — APPROVED** [review]
Tested. I didn't try every single return type on every single command, but those that I've tried work fine.

**Pull request merged**